umbraco/Umbraco-CMS
# Ability to change the culture of an existing language

Fixes #6503 

## Changes:

* Allow modifying a culture in the UI for an existing lang
* Allow modifying a culture for an existing lang in the controller
* Update lang repo to validate changing to an existing culture + tests
* Warn the user about expensive operation if they choose to change the culture of an existing language
* Change lang refresher to use payloads instead of just IDs
* When a lang is removed or has a culture change then lots of caches need to be evicted or rebuilt: Domains (all caches), IContent, nucache json DB store, nucache in memory, all indexes

## Testing

* Make sure tests pass + code review
* Have a 2-3 langs with variant content + domains assigned that render
* Change a language's properties but NOT the culture - you should not have a warning displayed
* Change a language's culture to one that already exists - you will get a warning, accept it, and then you will get a validation error
* Change a language's culture to one that doesn't exist - you will get a warning, accept it, it should save and return to the list showing the new language name
* Navigate to a variant content item that is under a node with domains assigned, cycle through each culture via the drop down and look at the Info tab/app to verify that all URLs are correct and routable
* Take note of this content ID and go to the Examine dashboard, Internal index, and search it with this ID. Click on the search result to show all of the fields in that document index, you will see fields that have culture suffixes, ensure that all culture suffixes are correct with the new culture suffix that you changed your lang too.
